### DISCUSSION TITLE
Animation plugin with slider?

### DISCUSSION THREAD

**Discussion by natehess** [jspsych/jsPsych#1480] [Q&A]
Hello,
I'm trying to add slider functionality identical to the slider and button in jspsych-video-slider-response.js to jspsych-animation.js, and running into performance issues. The slider and button work perfectly fine, but after a number of trials (5-10) I get bad frame rate drops in the canvas element. My animations are long, but performance is perfectly fine when using the default animation plugin. 

Currently I've added the slider and button code directly underneath the animation code. I've tried lots of variations, but performance remains the same.

Any tips?

Here's a link:
http://natehess.com/dynFractals/

I've gotten rid of some features so that you can just click "continue" to get through the stimuli quickly. The display rate starts dropping within 5-10 stimuli.

Thanks for the help!

**Comment by becky-gilbert** [comment]
I don't have a good answer to why the frame dropping is affected by the slider and button, so I'm hoping that someone else will respond...

However I did see that you're presenting images at a rate of 1 per frame, so just wanted to mention that it might be worth switching to `requestAnimationFrame` instead of `setInterval` to schedule the image presentations. The `setInterval` function is lower priority, so its function calls are more likely to be skipped/delayed if the browser's resources are limited. You can find `requestAnimationFrame` docs/examples [here](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) and [here](https://css-tricks.com/using-requestanimationframe/). And @kurokida's psychophysics plugin uses rAF, so you could check that out too: #1128

**Reply by kurokida** [reply]
Thank you for introducing my plugin.
But unfortunately, you can't get the slider to work with my plugin. 
To do that, you need to modify the plugin.

**Reply by natehess** [reply]
Thanks Becky! My next strategy was to try requestAnimationFrame, so I will let you know how it works out!

It turns out that I get this problem whenever I add any element outside the default jsPsych animation code, even a keypress listener. Fingers crossed that requestAnimationFrame does the trick.

**Reply by becky-gilbert** [reply]
Great, let us know if you want feedback on your rAF version or get it working!

**Reply by natehess** [reply]
Hi Becky,
Thought I'd post an update. The rAF approach worked really well and fixed the performance issue. I'm figuring out hosting this weekend, and will share a demo once everything is running smoothly.

**Reply by becky-gilbert** [reply]
@natehess thanks for the update, and very glad to hear that rAF fixed the problem! It'd be great to see the demo whenever it's ready.
